logstash-core/src/main/java/org/logstash/execution/WorkerLoop.java

```diff
@@ -62,13 +57,14 @@ public void run() {
                 readClient.closeBatch(batch);
                 if (isFlush) {
                     flushing.set(false);
+                    flushRequested.set(false);
                 }
-            } while (!shutdownRequested || isDraining());
+            } while (!shutdownRequested.get() || isDraining());
```

Comment:
@danhermann hmm maybe we should still save the result of `shutdownRequested.get()` to a normal local `boolean` and short-circuit calls to the `AtomicBoolean#get()` once it's true to convey that this is an "irreversible" thing? It would prevent some weird bugs ... maybe :)